hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/RowKeyGeneratorHelper.java

```diff
@@ -137,14 +137,15 @@ public static Object getNestedFieldVal(Row row, List<Integer> positions) {
     Object toReturn = null;
 
     while (index < totalCount) {
+      if (valueToProcess.isNullAt(positions.get(index))) {
```

Comment:
would you mind adding documentation at line 130 as to what this method is doing. I was the one who added his method, but now when I review it, took sometime to understand the full code ;)

Reply:
Good point.  I added the javadocs here to explain the logic.